### PR TITLE
Full angular removal

### LINKED ISSUE #108: Remove remaining legacy services
Remove Legacy Services from:
- [x] Multiple Historical Sessions Plugin
- [x] Data Product Cell
- [x] Stream Provider
- [x] LMST Time System
- [x] MSL Sol Time System
- [x] SCLK Time System
- [x] Venue Service

Testing Instructions
Regression w/ focus on above checklist

Areas to focus on:
Historical Session Selector
- make sure you can select sessions correctly and change sessions

Global Staleness
- set a `globalStalenessInterval` in the config, check for any errors when viewing telemetry

Data Products
- Verify you can "preview" data on a data product row

Time Systems
- Switch to various time systems and verify they work correctly and there are no errors

Venue Plugin
- enable `venueAware.enabled` in the config and provide a venue definition, there is an `ExamlpeVenueDefinitions.json` you can use an example.

Composition of variouis types
For the following types: https://github.com/NASA-AMMOS/openmct-mcws/pull/170/commits/a9a7fdca1c6022b76b85712434a1ffa349a0f388#diff-14394fd1e976c2fe95d68156e2071183636d09b660cbd914c92dd7a59ce72c5eR1-R19
Verify you cannot add any items to them. This can be done in the creation menu and making sure the location isn't available to be a parent and this can be done by dragging and dropping items into these views.

Export Data Action
For the following types: https://github.com/NASA-AMMOS/openmct-mcws/pull/170/commits/53a37ca0fd5d3c3f8ce5292c1f5cb87588dc10ec#diff-072e8c0aeb75e8c327e9421240a036913bbceb1ce0442b50cefa6c098cf6eba7R136-R144
Make sure you can "Export Historical Data" either by right clicking the item in the tree or by using the three dot menu in the view. Then verify that the data is exported correctly.

### PR DESCRIPTION
**Pull request by jvigliotta**
closes #108

This PR removes any remaining legacy code (openmct-legacy-support and angular mainly).

### CODE REVIEW

File: loader.js [hunk 1]
```diff
@@ -1,41 +1,31 @@
...
     './src/messagesView/plugin',
     './src/product-status/plugin',
     './about.html',
     './src/metadataAction/plugin',
     './src/clearDataIndicator/plugin',
-    './src/globalStaleness/plugin',
     './src/dictionaryView/plugin',
     './src/packetSummary/plugin',
     './src/containerView/plugin',
-    'openmct-legacy-support',
```
Comment (davetsay) [inline loader.js:18]: Daaaaayum

**Review by davetsay — CHANGES_REQUESTED** [review]
Wow good stuff. That's a bunch of work. I made some suggestions and had some questions.

- [x] I think we should just be consistent and remove the `this._openmct` references.
- [ ] We may want to file a follow up to improve venues, but should first find out how they really should work properly.
- [ ] Do any unit tests need updating?
- [ ] One thing i know was used in legacy repo, was the Data Product viewer, so let's test that.

File: src/exportDataAction/ExportDataAction.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    appliesTo(objectPath) {
+        const domainObject = objectPath[0];
+        const isValidType = this.validTypes.includes(domainObject.type);
+        if (!isValidType) {
+            return false;
+        }
+
+        const hasComposition = this._openmct.composition.get(domainObject) !== undefined;
```
[the remark this answers is not in the document]
Comment (davetsay) [reply src/exportDataAction/ExportDataAction.js:30]: So the plan is to add container type objects to the appliesTo if not there (tables, plots), and then to do a full check in invoke and let the user know if the data is non-exportable or not.

File: src/exportDataAction/ExportDataTask.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+     */
+    async invoke() {
+        const headers = [];
+        const headerSet = {};
+        const requestTelemetry = async (domainObject) => {
+            const telemetry = await this._openmct.telemetry.request(domainObject);
+
+            return telemetry;
+        };
+        const pullHeaders = (dataArray) => {
```
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/exportDataAction/ExportDataTask.js:31]: fo sho!

File: src/venues/Venue.js [hunk 4]
```diff
@@ -1,62 +1,59 @@
...
-        "packetSummaryEventStreamUrl",
-        "commandEventUrl",
-        "commandEventStreamUrl",
-        "messageStreamUrl",
-        "frameSummaryStreamUrl",
-    ];
+import SessionService from 'services/session/SessionService';
 
-    function Venue(configuration, openmct) {
+const DATASET_FIELDS = [
```
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/venues/Venue.js:3]: I pulled in what was in the constants from datasets and kept anything that wasn't included in that but existed here.

File: src/legacy/export/src/ExportDataTask.js [hunk 5]
```diff
@@ -1,72 +0,0 @@
...
-     */
-    ExportDataTask.prototype.perform = function () {
-        var headers = [],
-            headerSet = {},
-            exportService = this.exportService;
-
-        function issueRequest(domainObject) {
-            var telemetry = domainObject.getCapability('telemetry');
-            return telemetry ?
-                telemetry.requestData({ strategy: 'comprehensive' }) :
```
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/legacy/export/src/ExportDataTask.js:34]: I've re-added it to the new code.

**Comment by jvigliotta** [comment]
Follow up issues:
Update Tests: https://github.com/NASA-AMMOS/openmct-mcws/issues/172
Optimize Venue Pllugin: https://github.com/NASA-AMMOS/openmct-mcws/issues/173

File: src/venues/Venue.js [hunk 6]
```diff
@@ -1,59 +1,56 @@
...
-    'dataProductContentUrl',
-    'dataProductStreamUrl',
-    'packetUrl',
-    'packetContentUrl',
     'packetSummaryEventStreamUrl',
     'commandEventUrl',
     'commandEventStreamUrl',
-    'messageStreamUrl',
-    'frameSummaryStreamUrl',
+];
```
Comment (davetsay) [inline src/venues/Venue.js:12]: does some stuff need to be back ported into the constants file?

**Review by davetsay — CHANGES_REQUESTED** [review]
i think the valid types thing is the only thing that needs fixing

File: src/exportDataAction/ExportDataAction.js [hunk 7]
```diff
@@ -22,44 +22,64 @@ class ExportDataAction {
...
+        if (!this.isValidType(domainObject)) {
             return false;
         }
 
         const hasComposition = this.openmct.composition.get(domainObject) !== undefined;
-        const hasHistoricalTelemetry = this.openmct.telemetry.isTelemetryObject(domainObject) && 
-                !domainObject.telemetry.realtimeOnly;
+        const hasHistoricalTelemetry = this.hasHistoricalTelemetry(domainObject);
 
         return hasHistoricalTelemetry || !hasHistoricalTelemetry && hasComposition;
```
[the remark this answers is not in the document]
Comment (jvigliotta) [reply src/exportDataAction/ExportDataAction.js]: yep... my bad. Updated!

**Review by davetsay — APPROVED** [review]
Great work @jvigliotta . We finally did it!

**Pull request merged**